# Description: Fix build on non-Android ARM Linux builds without NEON.
#  - The common.gypi change allows us to disable NEON runtime detection with
#    (arm_neon==0 and arm_neon_optional==0) builds
#  - The sinc_resampler.cc changes add support for a mode where there is no
#    NEON support
#  - The isacfix.gypi changes fix the isac_neon target to not assume that
#    arm_version==7 means there is NEON support. This prevents a problem where
#    various lookup tables are defined twice (causing a link failure) with
#    (arm_neon==0 and arm_neon_optional==0 and arm_version==7) builds - once
#    in transform_neon.S (which shouldn't be built) and also in
#    transform_tables.c - with the latter being ifdef'd in correctly only when
#    both WEBRTC_ARCH_ARM_NEON and WEBRTC_DETECT_ARM_NEON aren't defined
# Author: Chris Coulson <chris.coulson@canonical.com>

diff --git a/third_party/webrtc/build/common.gypi b/third_party/webrtc/build/common.gypi
--- a/third_party/webrtc/build/common.gypi
+++ b/third_party/webrtc/build/common.gypi
@@ -278,17 +278,17 @@
         ],
         'conditions': [
           ['arm_version>=7', {
             'defines': ['WEBRTC_ARCH_ARM_V7',],
             'conditions': [
               ['arm_neon==1', {
                 'defines': ['WEBRTC_ARCH_ARM_NEON',],
               }],
-              ['arm_neon==0 and OS=="android"', {
+              ['arm_neon==0 and (OS=="android" or arm_neon_optional==1)', {
                 'defines': ['WEBRTC_DETECT_ARM_NEON',],
               }],
             ],
           }],
         ],
       }],
       ['target_arch=="mipsel" and mips_arch_variant!="r6" and android_webview_build==0', {
         'defines': [
diff --git a/third_party/webrtc/modules/audio_coding/codecs/isac/isacfix.gypi b/third_party/webrtc/modules/audio_coding/codecs/isac/isacfix.gypi
--- a/third_party/webrtc/modules/audio_coding/codecs/isac/isacfix.gypi
+++ b/third_party/webrtc/modules/audio_coding/codecs/isac/isacfix.gypi
@@ -75,17 +75,17 @@
         'fix/source/structs.h',
       ],
       'conditions': [
         ['OS!="win"', {
           'defines': [
             'WEBRTC_LINUX',
           ],
         }],
-        ['target_arch=="arm" and arm_version>=7', {
+        ['target_arch=="arm" and arm_version>=7 and (arm_neon==1 or arm_neon_optional==1)', {
           'dependencies': [ 'isac_neon', ],
           'sources': [
             'fix/source/lattice_armv7.S',
             'fix/source/pitch_filter_armv6.S',
           ],
           'sources!': [
             'fix/source/lattice_c.c',
             'fix/source/pitch_filter_c.c',
@@ -119,17 +119,17 @@
               ],
             }],
           ],
         }],
       ],
     },
   ],
   'conditions': [
-    ['target_arch=="arm" and arm_version>=7', {
+    ['target_arch=="arm" and arm_version>=7 and (arm_neon==1 or arm_neon_optional==1)', {
       'targets': [
         {
           'target_name': 'isac_neon',
           'type': 'static_library',
           'includes': ['../../../../build/arm_neon.gypi',],
           'dependencies': [
             '<(webrtc_root)/common_audio/common_audio.gyp:common_audio',
           ],
